kbase-extension/static/kbase/js/widgets/narrative_core/upload/stagingAreaViewer.js

```diff
@@ -97,6 +97,10 @@ define([
                 clearInterval(this.refreshInterval);
                 this.refreshInterval = undefined;
             }
+            if (this.fullDataTable) {
```

Comment:
We should clear the table out of memory (global? It's hard to say with jquery).
Also, the search extension function (used to filter down to the current directory) needs to be `pop`ped.